inference-engine/src/gna_plugin/memory/gna_memory_state.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gna_memory_state.hpp"
+#include "frontend/quantized_layer_params.hpp"
+#include "layer_transform.hpp"
+#include "preprocessing.hpp"
+#include "ie_layouts.h"
+
+namespace  GNAPluginNS {
+
+namespace memory {
+
+    std::string GNAMemoryState::GetName() const {
+        return name;
+    }
+
+    void GNAMemoryState::Reset() {
+        state->Reset();
+    }
+
+    InferenceEngine::Precision GNAMemoryState::getPrecision() const {
+        InferenceEngine::Precision state_precision;
+
+        if (state->getInput()) {
+            state_precision = state->getInput()->precision;
+        } else {
+            auto element_size = state->elementSizeBytes();
+            switch (element_size) {
+            case 4:
+                state_precision = InferenceEngine::Precision::FP32;
+                break;
+            case 2:
+                state_precision = InferenceEngine::Precision::I16;
+                break;
+            default:
+                THROW_GNA_EXCEPTION << "Incorrect state element size to determine precision";
```

Comment:
add element size value in message, also i would use extra name like state name since memoryLayers have names